server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1412,12 +1412,32 @@ private void closeConsistencyCheck() {
 
     if (!otherLogs.isEmpty() || !currentLogs.isEmpty() || !referencedLogs.isEmpty()) {
       String msg = "Closed tablet " + extent + " has walog entries in memory currentLogs = "
-          + currentLogs + "  otherLogs = " + otherLogs + " refererncedLogs = " + referencedLogs;
+          + currentLogs + "  otherLogs = " + otherLogs + " referencedLogs = " + referencedLogs;
       log.error(msg);
       throw new RuntimeException(msg);
     }
 
-    // TODO check lastFlushID and lostCompactID - ACCUMULO-1290
+    try {
+      long flushID = getFlushID();
+      if (lastFlushID != 0 && flushID == 0) {
+        String msg = "Closed tablet " + extent + " has not been flushed before being closed.";
+        log.error(msg);
+        throw new RuntimeException(msg);
+      }
+    } catch (NoNodeException e) {
+      e.printStackTrace();
```

Comment:
Didn't look at the rest of the PR, but one quick note: we'd want to log or throw these exceptions, rather than print them.